ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.core import Provenance
+from ctapipe.io.eventfilereader import EventFileReader
+from ctapipe.io.containers import DataContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+
+COUNT = 0
+
+
+class HessioFileReader(EventFileReader):
+
+    def __init__(self, config, tool, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            from pyhessio import open_hessio
+            from pyhessio import close_file
+            from pyhessio import HessioError
+            from pyhessio import HessioTelescopeIndexError
+            from pyhessio import HessioGeneralError
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
+        self.close_hessio = close_file
+        self.HessioError = HessioError
+        self.HessioTelescopeIndexError = HessioTelescopeIndexError
+        self.HessioGeneralError = HessioGeneralError
+
+        self.allowed_tels = None
+
+        global COUNT
+        if COUNT > 0:
+            self.log.warn("Only one pyhessio reader allowed at a time. "
+                          "Previous hessio file will be closed.")
+            self.close_hessio()
+        COUNT += 1
+
+    @staticmethod
+    def is_compatible(file_path):
+        return file_path.endswith('.gz')
+
+    @property
+    def camera(self):
+        return 'hessio'
+
+    def _generator(self):
```

Comment:
This can be `__iter__`, right?

Reply:
Ah, I see, this might create problems with the Only 1 file open constraint of pyhessio.

Reply:
yes, another reason why it will be nice to have a more modern format :-)

Reply:
Actually pyhessio would be fine with this as `__iter__` (as the `with` block is exited when a new `__iter__` is started. I had to move the generator from `__iter__` to its own function so I can keep reference to the generator. Therefore an operation like:
```
event = reader[3]
event = reader[3]
event = reader[4]
event = reader[2]
```
Continues from the last point and only starts from the beginning of the file again on the last line.

Reply:
Of course, if we kept a reference to the generator, it does have a problem with the 1-file pyhessio limit, that is why I recreate the reference the generator in `__iter__` (`reset`).